packages/cli/src/utils/checkpoint.ts

```diff
@@ -0,0 +1,177 @@
+import Debug from '@prisma/debug'
+import { getCLIPathHash, getConfig, getProjectHash, getSchema, parseEnvValue } from '@prisma/sdk'
+import type { Check } from 'checkpoint-client'
+import * as checkpoint from 'checkpoint-client'
+
+const debug = Debug('prisma:cli:checkpoint')
+
+/**
+ * Collect and prepare data then run the Checkpoint Client which will send some info to the remote Checkpoint Server
+ * It will be done in a child process so the CLI won't have to wait for it to finish to exit
+ *
+ * returns if the current CLI version is outdated.
+ *
+ * For more info about the data collected by the Checkpoint Server and how to disable it see:
+ * https://www.prisma.io/docs/concepts/more/telemetry
+ */
+export async function runCheckpointClientCheck({
+  schemaPath,
+  isPrismaInstalledGlobally,
+  version,
+  commandArray,
+  telemetryInformation,
+}: {
+  schemaPath: string
+  isPrismaInstalledGlobally: 'npm' | 'yarn' | false
+  version: string
+  commandArray: string[]
+  telemetryInformation: string
+}): Promise<Check.Result | 0> {
+  try {
+    // SHA256 identifier for the project based on the Prisma schema path
+    const projectPathHash = await getProjectHash()
+    // SHA256 of the cli path
+    const cliPathHash = getCLIPathHash()
+    // Redact the command options and make it a string
+    const commandAsString = redactCommandArray(commandArray).join(' ')
+    // Read schema and extract some data
+    const { schemaProvider, schemaPreviewFeatures, schemaGeneratorsProviders } = await tryToReadDataFromSchema(
+      schemaPath,
+    )
+
+    const data: Check.Input = {
+      // Name of the product
+      product: 'prisma',
+      // Currently installed version of the CLI
+      version,
+      // A unique hash of the path in which the CLI is installed
+      cli_path_hash: cliPathHash,
+      // A unique hash of the project's path, i.e.. the `schema.prisma`'s path
+      project_hash: projectPathHash,
+      // The first datasource provider (e.g. postgresql)
+      schema_providers: schemaProvider ? [schemaProvider] : undefined,
+      // previewFeatures from the prisma-client-js generator
+      schema_preview_features: schemaPreviewFeatures,
+      // Generator providers (e.g. prisma-client-js)
+      schema_generators_providers: schemaGeneratorsProviders,
+      // Type of CLI install: global or local
+      cli_install_type: isPrismaInstalledGlobally ? 'global' : 'local',
+      // Command with redacted options
+      command: commandAsString,
+      // Internal: Additional information from `--telemetry-information` option or `PRISMA_TELEMETRY_INFORMATION` env var
+      // Default: undefined
+      information: telemetryInformation || process.env.PRISMA_TELEMETRY_INFORMATION,
+      // Absolute CLI path
+      // Note: This won't be sent to the checkpoint server.
+      // TODO: Check if we can remove, probably not needed since cli_path_hash is defined
+      cli_path: process.argv[1],
```

Comment:
Is this something that can be checked now, or is it better we wait a future PR?

Reply:
I think we'll have the opportunity to do it later and merge this now